File: src/Features/NotifySignalsModel.cs
using System;
using Qml.Net;

namespace Features
{
    public class NotifySignalsModel
    {
        private string _bindableProperty = "";
        
        [NotifySignal]
        public string BindableProperty
        {
            get
            {
                return _bindableProperty;
            }
            set
            {
                if (_bindableProperty == value)
                {
                    // No signal is raised, Qml doesn't update any bound properties.
                    return;
                }

                _bindableProperty = value;
                this.ActivateSignal("bindablePropertyChanged");
            }
        }

        public void ChangeBindableProperty()
        {
            BindableProperty = DateTime.Now.ToLongTimeString();
        }
    }
}

File: src/Features.FSharp/NotifySignalsModel.fs
namespace Features

open System
open Qml.Net

type NotifySignalsModel() =
    let mutable bindableProperty = ""

    [<NotifySignal>]
    member this.BindableProperty
        with get() = bindableProperty
        and set(value) = 
            bindableProperty <- value
            this.ActivateSignal("bindablePropertyChanged") |> ignore
    
    member this.ChangeBindableProperty() = 
        this.BindableProperty <- DateTime.Now.ToLongTimeString();
